src/sections/aliases.tsx

```diff
@@ -0,0 +1,211 @@
+
+  return (
+    <div>
+      <input type="text" name="peerId" placeholder="peerId" onChange={handleChange} value={form.peerId} />
+      <input type="text" name="alias" placeholder="alias" onChange={handleChange} value={form.alias} />
+      <button onClick={() => {
+        if (loginData.ip && loginData.apiKey) {
+          dispatch(actionsAsync.setAliasThunk({ alias: form.alias, peerId: form.peerId, apiEndpoint: loginData.ip, apiToken: loginData.apiKey }))
+        }
+      }}>add</button>
+    </div>
+  )
+
+}
+
+
+/**
+ * Represents the expected structure of the parsed data.
+ */
+type ParsedData = {
+  [key: string]: string | number;
+}
+
+/**
+ * Props for the CSVUploader component.
+ */
+type CSVUploaderProps<T extends ParsedData> = {
+  /**
+   * Callback function called when the CSV data is successfully parsed.
+   * @param data The parsed data as an array of objects.
+   */
+  onParse: (data: T[]) => void;
+}
+
+/**
+ * Component for uploading and parsing CSV data.
+ */
+function CSVUploader<T extends ParsedData>({ onParse }: CSVUploaderProps<T>) {
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  /**
+   * Handles the file upload event.
+   * @param event The file upload event.
+   */
+  const handleFileUpload = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.[0];
+    const reader = new FileReader();
+
+    reader.onload = (e: ProgressEvent<FileReader>) => {
+      const contents = e.target?.result;
+      if (typeof contents === "string") {
+        parseCSV(contents);
+      }
+    };
+
+    if (file) {
+      reader.readAsText(file);
+    }
+  };
+
+  /**
+   * Parses the CSV content.
+   * @param csvContent The content of the CSV file.
+   */
+  const parseCSV = (csvContent: string) => {
+    const lines = csvContent.split("\n");
+    const parsedData: T[] = [];
+
+    // gets all keys, csv hold the headers on the first line
+    const header = lines[0].split(",");
+    const expectedObjectKeys = header.map((key) => key.trim());
+
+    // loop through each line, get the values and assign the value to the key
+    // then push the object to array parsedData
+    for (let i = 1; i < lines.length; i++) {
+      const values = lines[i].split(",");
+      if (values.length > 1) {
+        const dataObject: T = {} as T;
+
+        for (let j = 0; j < header.length; j++) {
+          const key = expectedObjectKeys[j];
+          const value = values[j]?.trim();
+          if (expectedObjectKeys.includes(key)) {
+            dataObject[key as keyof T] = value as T[keyof T];
+          }
+        }
+
+        parsedData.push(dataObject);
+      }
+    }
+
+    // after parsing run callback function
+    onParse(parsedData);
+
+    // Reset the file input value
+    if (fileInputRef.current) {
+      fileInputRef.current.value = "";
+    }
+  };
+
+  return (
+    <div>
+      import <input type="file" accept=".csv" ref={fileInputRef} onChange={handleFileUpload} placeholder="import" />
```

Comment:
Here just add an 'Import' button which will open the File Explorer for importing.
All that without the import text and 'no file chosen' etc.

Reply:
I can do this by hidding the <input /> and triggering it through a button but not sure how a button can open the file explorer without the <input />